### ISSUE TITLE
Using OpenAPI renders the generated output unusable

### ISSUE THREAD

**Issue by aabedraba** [fern-api/fern#1601]
In an internal hackathon I wanted to create a quick SDK. I already have an OpenAPI definition and wanted to use Fern to generate a TS SDK. 

I passed the following standard Todo OpenAPI: 

```json
{
  "openapi": "3.1.0",
  "info": {
    "title": "Todo List API",
    "description": "Example API, built using [Zuplo](https://zuplo.com/), for managing tasks within a todo list. You can find the source code for this example [here](https://github.com/zuplo/samples-todo-list-open-api).",
    "version": "1.0.0"
  },
  "tags": [
    {
      "name": "Standard Todo List Endpoints",
      "description": "Standard endpoints for managing todos list items."
    },
    {
      "name": "Extended Todo List Endpoints",
      "description": "Extended endpoints for managing todos list items. The endpoints are mainly for convenience, performing common operations on top of the standard endpoints."
    }
  ],
  "components": {
    "schemas": {
      "TodoObject": {
        "type": "object",
        "required": [
          "id",
          "userId",
          "title",
          "completed"
        ],
        "properties": {
          "id": {
            "type": "string"
          },
          "userId": {
            "type": "string"
          },
          "title": {
            "type": "string"
          },
          "completed": {
            "type": "boolean"
          }
        },
        "examples": [
          {
            "userId": 1,
            "id": 1,
            "title": "Take out the trash",
            "completed": false
          }
        ]
      },
      "AnonymousTodoObject": {
        "type": "object",
        "required": [
          "id",
          "title",
          "completed"
        ],
        "properties": {
          "id": {
            "type": "string"
          },
          "title": {
            "type": "string"
          },
          "completed": {
            "type": "boolean"
          }
        },
        "examples": [
          {
            "id": 1,
            "title": "Take out the trash",
            "completed": false
          }
        ]
      },
      "TodoListObject": {
        "type": "array",
        "items": {
          "$ref": "#/components/schemas/TodoObject"
        },
        "examples": [
          [
            {
              "userId": 1,
              "id": 1,
              "title": "Take out the trash",
              "completed": false
            }
          ]
        ]
      },
      "InsertTodoObject": {
        "type": "object",
        "title": "Todo Insert Schema",
        "required": [
          "userId",
          "title",
          "completed"
        ],
        "additionalProperties": false,
        "properties": {
          "userId": {
            "type": "integer",
            "description": "The userId that created the todo list item.",
            "examples": [
              1
            ]
          },
          "title": {
            "type": "string",
            "description": "The title of the todo list item.",
            "examples": [
              "Wash the dishes"
            ]
          },
          "completed": {
            "type": "boolean",
            "description": "Whether or not the todo list item is completed.",
            "examples": [
              false
            ]
          }
        },
        "examples": [
          {
            "userId": 1,
            "title": "Wash the dishes",
            "completed": false
          }
        ]
      },
      "UpdateTodoObject": {
        "type": "object",
        "title": "Update Todo Object",
        "additionalProperties": false,
        "required": [
          "userId",
          "completed",
          "title"
        ],
        "properties": {
          "userId": {
            "type": "integer",
            "description": "The userId that created the todo list item.",
            "examples": [
              1
            ]
          },
          "title": {
            "type": "string",
            "description": "The title of the todo list item.",
            "examples": [
              "Make dinner"
            ]
          },
          "completed": {
            "type": "boolean",
            "description": "Whether or not the todo list item is completed.",
            "examples": [
              false
            ]
          }
        },
        "examples": [
          {
            "userId": 1,
            "title": "New Title",
            "completed": false
          }
        ]
      },
      "SchemaValidationError": {
        "type": "object",
        "required": [
          "type",
          "title",
          "status",
          "detail",
          "instance"
        ],
        "properties": {
          "type": {
            "type": "string"
          },
          "title": {
            "type": "string"
          },
          "status": {
            "type": "number"
          },
          "detail": {
            "type": "string"
          },
          "instance": {
            "type": "string"
          },
          "trace": {
            "type": "object"
          },
          "errors": {
            "type": "array",
            "items": {
              "type": "string"
            },
            "minItems": 1
          }
        },
        "examples": [
          {
            "type": "https://httpproblems.com/http-status/400",
            "title": "Bad Request",
            "status": 400,
            "detail": "Incoming body did not pass schema validation",
            "instance": "/v1/todos",
            "trace": {
              "timestamp": "2023-02-27T18:53:05.997Z",
              "requestId": "b1e1c2a9-da7b-436c-aa89-2f78918047b2",
              "buildId": "83e3d0f1-89a8-46ea-b040-e0a2432f2ea5",
              "rayId": "7a031f102747944d-SJC"
            },
            "errors": [
              "Body must have required property 'userId'"
            ]
          }
        ]
      },
      "UserObject": {
        "type": "object",
        "properties": {
          "id": {
            "type": "number"
          },
          "name": {
            "type": "string"
          },
          "username": {
            "type": "string"
          },
          "email": {
            "type": "string"
          },
          "address": {
            "type": "object",
            "properties": {
              "street": {
                "type": "string"
              },
              "suite": {
                "type": "string"
              },
              "city": {
                "type": "string"
              },
              "zipcode": {
                "type": "string"
              },
              "geo": {
                "type": "object",
                "properties": {
                  "lat": {
                    "type": "string"
                  },
                  "lng": {
                    "type": "string"
                  }
                }
              }
            }
          },
          "phone": {
            "type": "string"
          },
          "website": {
            "type": "string"
          },
          "company": {
            "type": "object",
            "properties": {
              "name": {
                "type": "string"
              },
              "catchPhrase": {
                "type": "string"
              },
              "bs": {
                "type": "string"
              }
            }
          }
        },
        "examples": [
          {
            "id": 1,
            "name": "Leanne Graham",
            "username": "Bret",
            "email": "Sincere@april.biz",
            "address": {
              "street": "Kulas Light",
              "suite": "Apt. 556",
              "city": "Gwenborough",
              "zipcode": "92998-3874",
              "geo": {
                "lat": "-37.3159",
                "lng": "81.1496"
              }
            },
            "phone": "1-770-736-8031 x56442",
            "website": "hildegard.org",
            "company": {
              "name": "Romaguera-Crona",
              "catchPhrase": "Multi-layered client-server neural-net",
              "bs": "harness real-time e-markets"
            }
          }
        ]
      }
    }
  },
  "paths": {
    "/v1/todos": {
      "x-zuplo-path": {
        "pathMode": "open-api"
      },
      "get": {
        "tags": [
          "Standard Todo List Endpoints"
        ],
        "summary": "Get all todos",
        "description": "Gets all the todos in the todo list.",
        "operationId": "b61c0cd1-b380-4440-a430-840ea85f3e9c",
        "responses": {
          "200": {
            "description": "A list of todos",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/TodoListObject"
                }
              }
            }
          }
        },
        "x-zuplo-route": {
          "corsPolicy": "none",
          "handler": {
            "export": "urlRewriteHandler",
            "module": "$import(@zuplo/runtime)",
            "options": {
              "rewritePattern": "https://jsonplaceholder.typicode.com/todo",
              "forwardSearch": true
            }
          },
          "policies": {
            "inbound": []
          }
        }
      },
      "post": {
        "summary": "Create Todo",
        "description": "Creates a todo list item.",
        "x-zuplo-route": {
          "corsPolicy": "none",
          "handler": {
            "export": "urlRewriteHandler",
            "module": "$import(@zuplo/runtime)",
            "options": {
              "rewritePattern": "https://jsonplaceholder.typicode.com/todos",
              "forwardSearch": true
            }
          },
          "policies": {
            "inbound": [
              "validate-json-schema-inbound"
            ]
          }
        },
        "tags": [
          "Standard Todo List Endpoints"
        ],
        "operationId": "f9e30d74-56ca-4f1e-bcb3-75fe305ea5e4",
        "requestBody": {
          "description": "Payload required to create a todo list item.",
          "required": true,
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/InsertTodoObject"
              }
            }
          }
        },
        "responses": {
          "201": {
            "description": "The created todo list item",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/TodoObject"
                }
              }
            }
          },
          "400": {
            "description": "Schema validation error",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/SchemaValidationError"
                }
              }
            }
          }
        }
      }
    },
    "/v1/$todosAndUsers": {
      "x-zuplo-path": {
        "pathMode": "open-api"
      },
      "get": {
        "summary": "Get todos with user information",
        "description": "Gets the list of todos, with a `user` property that contains information about the user that created the todo list item.",
        "x-zuplo-route": {
          "corsPolicy": "none",
          "handler": {
            "export": "default",
            "module": "$import(./modules/todos-and-users)",
            "options": {}
          },
          "policies": {
            "inbound": []
          }
        },
        "operationId": "398f226b-7401-4906-9aba-8dee92068f94",
        "tags": [
          "Extended Todo List Endpoints"
        ],
        "responses": {
          "200": {
            "description": "Todos with creating user data attached",
            "content": {
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "object",
                    "required": [
                      "user",
                      "userId",
                      "id",
                      "title",
                      "completed"
                    ],
                    "allOf": [
                      {
                        "$ref": "#/components/schemas/TodoObject"
                      }
                    ],
                    "properties": {
                      "user": {
                        "$ref": "#/components/schemas/UserObject"
                      }
                    },
                    "examples": [
                      {
                        "userId": 1,
                        "id": 1,
                        "title": "Take out the trash",
                        "completed": false,
                        "user": {
                          "id": 1,
                          "name": "Leanne Graham",
                          "username": "Bret",
                          "email": "Sincere@april.biz",
                          "address": {
                            "street": "Kulas Light",
                            "suite": "Apt. 556",
                            "city": "Gwenborough",
                            "zipcode": "92998-3874",
                            "geo": {
                              "lat": "-37.3159",
                              "lng": "81.1496"
                            }
                          },
                          "phone": "1-770-736-8031 x56442",
                          "website": "hildegard.org",
                          "company": {
                            "name": "Romaguera-Crona",
                            "catchPhrase": "Multi-layered client-server neural-net",
                            "bs": "harness real-time e-markets"
                          }
                        }
                      }
                    ]
                  }
                }
              }
            }
          }
        }
      }
    },
    "/v1/$todosWithNoUserId": {
      "x-zuplo-path": {
        "pathMode": "open-api"
      },
      "get": {
        "summary": "Get todos with user information removed",
        "description": "Gets the list of todos with the user information stripped out. Use this API if the user's ID is sensitive information that can't be leaked to the product.",
        "x-zuplo-route": {
          "corsPolicy": "none",
          "handler": {
            "export": "urlRewriteHandler",
            "module": "$import(@zuplo/runtime)",
            "options": {
              "rewritePattern": "https://jsonplaceholder.typicode.com/todos",
              "forwardSearch": true
            }
          },
          "policies": {
            "inbound": [],
            "outbound": [
              "custom-code-outbound"
            ]
          }
        },
        "operationId": "d98a939d-d1a2-4352-8c45-3566bfd67860",
        "tags": [
          "Extended Todo List Endpoints"
        ],
        "responses": {
          "200": {
            "description": "Anonymized todo list",
            "content": {
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/AnonymousTodoObject"
                  }
                }
              }
            }
          }
        }
      }
    },
    "/v1/todos/:todoId": {
      "x-zuplo-path": {
        "pathMode": "open-api"
      },
      "patch": {
        "summary": "Update Todo",
        "description": "Updates a todo list item with a matching `todoId`. Will return an error if a matching todo item is not found.",
        "x-zuplo-route": {
          "corsPolicy": "none",
          "handler": {
            "export": "urlRewriteHandler",
            "module": "$import(@zuplo/runtime)",
            "options": {
              "rewritePattern": "https://jsonplaceholder.typicode.com/todos/${params.todoId}",
              "forwardSearch": true
            }
          },
          "policies": {
            "inbound": [
              "validate-json-schema-inbound-1"
            ]
          }
        },
        "operationId": "f3334d8b-37f9-489b-87c5-08a8beb5657c",
        "tags": [
          "Standard Todo List Endpoints"
        ],
        "parameters": [
          {
            "name": "todoId",
            "in": "path",
            "description": "ID of the todo list item to be modified.",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "The new todo object.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/TodoObject"
                }
              }
            }
          },
          "400": {
            "description": "Schema validation error",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/SchemaValidationError"
                }
              }
            }
          }
        },
        "requestBody": {
          "description": "Request body to update a todo list object.",
          "required": true,
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/UpdateTodoObject"
              }
            }
          }
        }
      },
      "delete": {
        "summary": "Delete Todo",
        "description": "Deletes a todo list item. Will return an error if the todo list item does not exist.",
        "x-zuplo-route": {
          "corsPolicy": "none",
          "handler": {
            "export": "urlRewriteHandler",
            "module": "$import(@zuplo/runtime)",
            "options": {
              "rewritePattern": "https://jsonplaceholder.typicode.com/todos/${params.todoId}",
              "forwardSearch": true
            }
          },
          "policies": {
            "inbound": []
          }
        },
        "operationId": "1647d06c-2a96-41ab-a2f7-ebb55d5bcd76",
        "tags": [
          "Standard Todo List Endpoints"
        ],
        "parameters": [
          {
            "name": "todoId",
            "in": "path",
            "description": "ID of the todo list item to be deleted.",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Empty response.",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "examples": [
                    {}
                  ]
                }
              }
            }
          }
        }
      }
    }
  }
}
```

Running `fern init --openapi routes.oas.json` and then `fern generate` results in the following error: 

```sh
[api]: extendedTodoListEndpoints.yml -> types -> _InlinedType0
       Type name must begin with a letter
[api]: extendedTodoListEndpoints.yml -> types -> _InlinedType0 -> extends -> Standard Todo List Endpoints.TodoObject
       Type Standard Todo List Endpoints.TodoObject is not defined.
[api]: extendedTodoListEndpoints.yml -> types -> _InlinedType0 -> properties -> user
       Type __package__.UserObject is not defined.
[api]: standardTodoListEndpoints.yml -> service -> endpoints -> 1647D06C2A9641AbA2F7Ebb55D5Bcd76
       Path parameter is unreferenced in endpoint: todoId.
[api]: standardTodoListEndpoints.yml -> service -> endpoints -> f3334D8B37F9489B87C508A8Beb5657C
       Path parameter is unreferenced in endpoint: todoId.
```

Not sure if it would be a good idea to continue from here, because the next time I update my OpenAPI the fixes would be overridden and I'd have to do it all over again... 

Any recommendations?

**Comment by dsinghvi** [comment]
@aabedraba the command `fern init --openapi <path>` is a one time import from OpenAPI into fern. Instead if you setup your project like by passing the openapi directly then 1) the errors should go away and 2) you'll be update easily as your OpenAPI changes. 
<img width="850" alt="Screen Shot 2023-05-17 at 8 19 34 AM" src="https://github.com/fern-api/fern/assets/10870189/c8538c6b-f6bb-407a-a108-dff0b2aec816">

**Comment by dsinghvi** [comment]
@aabedraba thanks for reporting this issue. I've just released [0.9.0](https://github.com/fern-api/fern/releases/tag/0.9.0) of the CLI with the fix. 

Now if you run `fern init --openapi <workspace>` then Fern will keep your [OpenAPI](https://docs.buildwithfern.com/spec/openapi) instead of converting it and the issues complaining about `_InlinedType` will also go away.

**Comment by dsinghvi** [comment]
@aabedraba closing this issue, feel free to reopen if it still doesn't work for you
